File: mobile/lib/features/reviews/domain/entities/review.dart
class Review {
  const Review({
    required this.id,
    required this.bookingId,
    required this.customerId,
    required this.vendorId,
    required this.rating,
    required this.text,
    required this.createdAt,
  });

  final String id;
  final String bookingId;
  final String customerId;
  final String vendorId;
  final int rating;
  final String text;
  final String createdAt;
}

File: backend/services/core/internal/domain/review.go
package domain

import "time"

// Review is a customer's evaluation of a vendor after a completed booking.
type Review struct {
	ID         string    `json:"id"`
	BookingID  string    `json:"bookingId"`
	CustomerID string    `json:"customerId"`
	VendorID   string    `json:"vendorId"`
	Rating     int       `json:"rating"`
	Text       string    `json:"text"`
	CreatedAt  time.Time `json:"createdAt"`
}

// MinRating and MaxRating bound the 5-star scale.
const (
	MinRating = 1
	MaxRating = 5
)

// ValidRating reports whether the rating falls within the allowed range.
func ValidRating(r int) bool { return r >= MinRating && r <= MaxRating }
